doc/source/tune.rst

```diff
@@ -0,0 +1,160 @@
+
+.. image:: ray-tune-viskit.png
+
+Finally, to view the results with a `parallel coordinates visualization <https://en.wikipedia.org/wiki/Parallel_coordinates>`__, open `ParalleCoordinatesVisualization.ipynb <https://github.com/ray-project/ray/blob/master/python/ray/tune/ParallelCoordinatesVisualization.ipynb>`__ as follows and run its cells:
+
+::
+
+    $ cd $RAY_HOME/python/ray/tune
+    $ jupyter-notebook ParallelCoordinatesVisualization.ipynb
+
+Trial Variant Generation
+------------------------
+
+In the above example, we specified a grid search over two parameters using the ``grid_search`` helper function. Ray.tune also supports sampling parameters from user-specified lambda functions, which can be used in combination with grid search.
+
+The following shows grid search over two nested parameters combined with random sampling from two lambda functions. Note that the value of ``beta`` depends on the value of ``alpha``, which is represented by referencing ``spec.config.alpha`` in the lambda function. This lets you specify conditional parameter distributions.
+
+::
+
+    "config": {
+        "alpha": lambda spec: np.random.uniform(100),
+        "beta": lambda spec: spec.config.alpha * np.random.normal(),
+        "nn_layers": [
+            grid_search([16, 64, 256]),
+            grid_search([16, 64, 256]),
+        ],
+    },
+    "repeat": 10,
+
+By default, each random variable and grid search point is sampled once. To take multiple random samples or repeat grid search runs, add ``repeat: N`` to the experiment config. E.g. in the above, ``"repeat": 10`` repeats the 3x3 grid search 10 times, for a total of 90 trials, each with randomly sampled values of ``alpha`` and ``beta``.
+
+For more information on variant generation, see `variant_generator.py <https://github.com/ray-project/ray/blob/master/python/ray/tune/variant_generator.py>`__.
+
+Early Stopping
+--------------
+
+To reduce costs, long-running trials can often be early stopped if their initial performance is not promising. Ray.tune allows early stopping algorithms to be plugged in on top of existing grid or random searches. This can be enabled by setting the ``scheduler`` parameter of ``run_experiments``, e.g.
+
+::
+
+    run_experiments({...}, scheduler=MedianStoppingRule())
+
+Currently we support the following early stopping algorithms, or you can write your own that implements the `TrialScheduler <https://github.com/ray-project/ray/blob/master/python/ray/tune/trial_scheduler.py>`__ interface:
+
+.. autoclass:: ray.tune.median_stopping_rule.MedianStoppingRule
+.. autoclass:: ray.tune.hyperband.HyperBandScheduler
+
+Checkpointing support
+---------------------
+
+To enable checkpoint / resume, the full ``Trainable`` API must be implemented. This is required to support resource multiplexing in schedulers such as HyperBand. For example, all `RLlib agents <https://github.com/ray-project/ray/blob/master/python/ray/rllib/agent.py>`__ implement the ``Trainable`` API.
```

Comment:
On another note, it is not very clear if you still need to provide the "reporter" interface if you use the Trainable API. More specifically, Line 9-12 of `trainable.py` should sit somewhere here in the documentation

Reply:
Done